### PR TITLE
Close the Loop: annotating lanes in potree

### PR DESCRIPTION
**Pull request by nmhaddadND**
This PR finishes closing the loop between potree and AWS by adding a AWS Lambda function and writing a new `lanes.fb` file directly to S3 instead of manually calling a script and uploading the new file. The script is modified and placed in a Lambda function, called by potree. There are no known bugs introduced by this PR, but versioning will need to be used for our Lambda function if we intend to make changes to it. This is a best practice. 

Location of Lambda function:
https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/UpdateLanes?tab=configuration

### CODE REVIEW

File: demo/paramLoader.js [hunk 1]
```diff
@@ -54,6 +54,14 @@ export const s3 = bucket && region && name && accessKeyId && secretAccessKey &&
...
     });
 
+export const lambda = new AWS.Lambda({
+  region: region,
+  accessKeyId: accessKeyId,
+  secretAccessKey: secretAccessKey,
+  sessionToken: sessionToken,
+  apiVersion: '2015-03-31'
+});
+
```
Comment (bcasselsND) [inline demo/paramLoader.js:64]: Yuck. Let's be getting rid of global variables, not adding more. Make a function that returns a new Lambda object.

File: common/playbar.js [hunk 2]
```diff
@@ -413,41 +413,41 @@ async function saveLaneChanges () {
...
-
-  const schemaUrl = s3.getSignedUrl('getObject', {
-    Bucket: bucket,
-    Key: `${name}/5_Schemas/GroundTruth_generated.js`
-  });
-  const FlatbufferModule = await import(schemaUrl);
-  await saveLaneChangesHelper(lane, FlatbufferModule);
-  const bytes = await createLanesFlatbuffer(lane, FlatbufferModule);
-  writeFileToS3(s3, bucket, name, "2_Truth", "upload-testing-lanes.fb", bytes);
+  await updateSpine(bucket, name, lane.left, lane.right);
```
Comment (bcasselsND) [inline common/playbar.js:431]: `updateSpine` isn't `async`, so no need to `await`.

File: common/playbar.js [hunk 3]
```diff
@@ -413,41 +413,41 @@ async function saveLaneChanges () {
...
-  lane.rightPointAnnotationStatus = Array.from({ length: rightLength }).map(x => 1);
+function updateSpine (bucket, name, left, right) {
+  const input = {
+    bucket: bucket,
+    name: name,
+    left: left,
+    right: right
+  };
+  lambda.invoke({
+    FunctionName: 'arn:aws:lambda:us-east-1:757877321035:function:UpdateLanes',
```
Comment (bcasselsND) [inline common/playbar.js:442]: Can a lambda in us-east-1 access S3 in other regions? Is it okay to just have the one in us-east-1?

File: common/playbar.js (hunk 2 above)
Comment (nmhaddadND) [reply common/playbar.js:431]: I meant to use await inside of update spine to wait for the lambda function to finish before proceeding. Those changes are fixed.

File: common/playbar.js (hunk 3 above)
Comment (nmhaddadND) [reply common/playbar.js:442]: Yes. The problem we may run into is slight latency between data centers.

File: common/playbar.js [hunk 4]
```diff
@@ -431,23 +431,24 @@ async function saveLaneChanges () {
...
+async function updateSpine (bucket, name, left, right) {
   const input = {
     bucket: bucket,
     name: name,
     left: left,
     right: right
   };
-  lambda.invoke({
+  const lambda = getLambda();
+  await lambda.invoke({
```
Comment (bcasselsND) [inline common/playbar.js:442]: I see no promise that you are awaiting. Perhaps you meant to [turn the callback into a promise](https://zellwk.com/blog/converting-callbacks-to-promises/)?

Comment (bcasselsND) [reply common/playbar.js:442]: BTW, if you make `updateSpine` return a promise, then it does not need to itself be `async`. Functions only need to be `async` if they, themselves, use `await`.

Comment (nmhaddadND) [reply common/playbar.js]: Removed all that junk.

**Pull request merged**